components/camel-sjms2/src/main/java/org/apache/camel/component/sjms2/Sjms2Component.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.sjms2;
+
+import java.util.Map;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.sjms.SjmsComponent;
+import org.apache.camel.component.sjms.SjmsEndpoint;
+
+/**
+ * The <a href="http://camel.apache.org/sjms">Simple JMS2</a> component.
+ */
+public class Sjms2Component extends SjmsComponent {
+
+    public Sjms2Component() {
+        super(SjmsEndpoint.class);
+    }
+
+    @Override
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+        validateMepAndReplyTo(parameters);
+        Sjms2Endpoint endpoint = new Sjms2Endpoint(uri, this, remaining);
```

Comment:
It would likely be easier to create a protected method in the camel-sjms component to create the endpoint class, we do that for some other components we extend. Then this component just need to create a new Sjms2Endpoint and set the jms2 object factory and all the other configuration below is the same and done on the base class